client/hooks/useRegexText.ts

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect } from 'react';
+
+const useRegexText = (state: string, regex: RegExp | string, text: { default: string, match: string, unMatch: string }): string => {
+    const [regexText, setRegexText] = useState(text.default);
+  
+    useEffect(() => {
+      if (regex instanceof RegExp)  {
+        if (state !== '' && !regex.test(state)) {
+          setRegexText(text.unMatch);
+        } 
+        else if (state !== '' && regex.test(state)){
+          setRegexText(text.match);
+        }
+      }
+
+      else if (typeof regex === 'string') {
+        if (state !== '' && regex !== '' && state !== regex) {
+          setRegexText(text.unMatch);
+        } 
+        else if (state !== '' && regex !== '' && state === regex) {
+          setRegexText(text.match);
+        }
+      }
+    }, [state, regex, text]);
```

Comment:
공통되는 조건과 분기를 조금 더 정리하면 가독성이 더 좋을 것 같아요.
또, state가 빈값인경우는 default값으로 돌려주지는 않아도 괜찮은건가요?

```suggestion
    useEffect(() => {
      if(!state){
        setRegexText(text.default)
        return;
      }
      if (regex instanceof RegExp)  {
        setRegexText(regex.test(state) ? text.match : text.unMatch)
      }

      else if (regex && typeof regex === 'string') {
        setRegexText(state === regex ? text.match : text.unMatch)
      }
    }, [state, regex, text]);
```

하나 더 조언드리면 regex를 string으로 받아서 값과 비교하는것은 조금 이상한 것 같아요(실제로 regex타입이 아니니까요). 별도의 param으로 처리하는게 더 가독성이 좋을 듯 합니다.

Reply:
state가 빈칸일 땐 pram text의 default 값으로 반환하게 됩니다!
말씀해주시니 공통된 분기가 3개나 보이네요 ... 감사합니다.
비밀번호 동일성 체크하는 것은 고민하다 아예 훅에서 뺄까 고민했었는데  부분인데 별도의 param으로 따로 둘 생각을 못했네요
수정하겠습니다!